src/aktualizr_secondary/aktualizr_secondary.cc

```diff
@@ -78,146 +80,61 @@ int32_t AktualizrSecondary::getRootVersionResp(bool director) const {
   return Uptane::extractVersionUntrusted(root_meta);
 }
 
-bool AktualizrSecondary::putRootResp(const std::string& root, bool director) {
+bool AktualizrSecondary::putRoot(const std::string& root, bool director) {
   (void)root;
   (void)director;
   LOG_ERROR << "putRootResp is not implemented yet";
   return false;
 }
 
-bool AktualizrSecondary::sendFirmwareResp(const std::shared_ptr<std::string>& firmware) {
-  auto targetsForThisEcu = director_repo_.getTargets(getSerial(), getHardwareID());
+bool AktualizrSecondary::putMetadata(const Metadata& metadata) { return doFullVerification(metadata); }
 
-  if (targetsForThisEcu.size() != 1) {
-    LOG_ERROR << "Invalid number of targets (should be one): " << targetsForThisEcu.size();
+bool AktualizrSecondary::sendFirmware(const std::string& firmware) {
+  // TODO: how to handle the case when secondary is rebooted after metadata are received
+  if (!pending_target_.IsValid()) {
+    LOG_ERROR << "No any pending target to receive update data/image for";
```

Comment:
This message is also a bit confusing. Maybe change to "Aborting image download; no valid target found."

Reply:
ok